sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -2521,10 +2508,280 @@ public void testDetermineCharsetFromMessageASCIISpecialCase(@TempDir final File
                 dir, null, null, StandardCharsets.ISO_8859_1, StandardCharsets.UTF_8, "application/soap+xml");
         // then
         assertEquals(StandardCharsets.ISO_8859_1, actualCharset);
-        Mockito.verify(this.testRunObserver, VerificationModeFactory.atLeastOnce())
-                .invalidateTestRun(anyString());
+        verify(this.testRunObserver, VerificationModeFactory.atLeastOnce()).invalidateTestRun(anyString());
+    }
+
+    /**
+     * Checks that convertMessageToMessageContent() works when it does not detect any decoding problems in the message.
+     * @param dir - a temporary directory.
+     * @throws IOException - when something goes wrong.
+     */
+    @Test
+    public void testConvertToMessageContent(@TempDir final File dir) throws IOException {
+
+        final Charset charsetInHttpHeader = StandardCharsets.UTF_8;
+        final Charset charsetInXMLDeclaration = StandardCharsets.UTF_8;
+        final Charset charsetInXMLDeclarationEncoding = StandardCharsets.UTF_8;
```

Comment:
I suggest to change to an encoding other than the default/fallback, at best one that would throw an error with the default.

Reply:
Unfortunately, using anything but UTF-8 here causes the TestRun to be invalidates due to MDPWS:R0007_0 and would hence not constitute a good case.

Reply:
I tried creating another test case to cover the case, but found that it is impossible. The case is impossible to reach and consequently there is no control flow path for that case that could be tested.

Reply:
> The case is impossible to reach and consequently there is no control flow path for that case that could be tested.

If that would be true, the mentioned part of the code would be unnecessary and could be removed.
However I do not agree with this assessment, it is reachable as some of the already exisiting unit tests proof.

Reply:
Misunderstanding resolved, Test case added.